test/functional/p2p_invalid_block.py

```diff
@@ -133,5 +136,19 @@ def run_test(self):
         self.log.info("Test inflation by duplicating input")
         peer.send_blocks_and_test([block4], node, success=False,  reject_reason='bad-txns-inputs-duplicate')
 
+
+        self.log.info("Test accepting identical block after rejecting it due to a future timestamp.")
+
+        t = int(time.time())
+        node.setmocktime(t)
+        block = create_block(tip, create_coinbase(height), t + 7201)
+        block.hashMerkleRoot = block.calc_merkle_root()
+        block.solve()
+        # NOTE: Need force_send because the block will get rejected without a getdata otherwise
+        node.p2ps[0].send_blocks_and_test([block], node, force_send=True, success=False, reject_reason='time-too-new')
+
+        node.setmocktime(t + 1)
+        node.p2ps[0].send_blocks_and_test([block], node, success=True, timeout=5)
```

Comment:
I'd prefer referencing the p2p object directly instead of through the node's p2p list, similar to the code above. If a subtest is added that connects/disconnects peers, this index could no longer refer to the peer you want.

```suggestion
        peer.send_blocks_and_test([block], node, force_send=True, success=False, reject_reason='time-too-new')

        node.setmocktime(t + 1)
        peer.send_blocks_and_test([block], node, success=True, timeout=5)
```

Reply:
Thanks, I agree this is both clearer and better.